classes/omnect_bootloader_versioning.bbclass

```diff
@@ -0,0 +1,69 @@
+python  do_bootloader_checksum() {
+    import glob
+    import hashlib
+
+    checksum_files = d.getVar("OMNECT_BOOTLOADER_CHECKSUM_FILES").split(" ")
+    checksum_files_ignore = []
+    checksum_files_ignore_str = d.getVar("OMNECT_BOOTLOADER_CHECKSUM_FILES_GLOB_IGNORE")
+    if checksum_files_ignore_str:
+        checksum_files_ignore = checksum_files_ignore_str.split(" ")
+
+    used_checksum_file_list = []
+    checksum_list = []
+
+    for checksum_file in checksum_files:
+        checksum_glob_files = glob.glob(checksum_file)
+        if checksum_file and not checksum_glob_files:
+            bb.fatal("couldn't handle checksum_file: %s" % checksum_file)
+
+        bb.debug(2, "checksum_file: %s, checksum_glob_files: %s" % (checksum_file, checksum_glob_files))
+        for checksum_glob_file in checksum_glob_files:
+            if checksum_glob_file in checksum_files_ignore:
+                continue
+            try:
+                with open(checksum_glob_file, "rb") as f:
+                    digest = hashlib.file_digest(f, "sha256")
+                    used_checksum_file_list.append(checksum_glob_file)
+                    checksum_list.append(digest.digest())
+            except OSError:
+                bb.fatal("Unable to open \"%s\"" % (checksum_glob_file))
+
+    m = hashlib.sha256()
+    for checksum in checksum_list:
+        m.update(checksum)
+    version_checksum = m.hexdigest()
+
+    # check if we should override version_checksum
+    version_checksum_override = d.getVar("OMNECT_BOOTLOADER_CHECKSUM_COMPATIBLE")
+    if version_checksum_override:
+        version_checksum_override_list = version_checksum_override.split(" ",1)
+        if version_checksum_override_list[0] == version_checksum:
+            bb.debug(1, "overriding version_checksum %s -> %s" % (version_checksum,version_checksum_override_list[1]))
+            version_checksum = version_checksum_override_list[1]
```

Comment:
I orga a refinement for clarification

Reply:
yes, it would be a long string indeed which cannot be compared by simply looking at them.
but making them space separated entries of the format <filepath>:<shasum> would allow to separate them easily in a list which could be sorted and placed into a list which in turn could be simply diff'ed between builds.
that was my spontaneous idea, maybe there are more elegant ways, but I do see the necessity coming to find out why on earth the u-boot version seems to have changed when we didn't expect that ;-)
let's continue that in the refinement as necessary, though.

Reply:
decision: write to own txt file via deploy mechanism
list of:
"filename<space>hash"

Reply:
done